src/pages/test-report/cat-a-mod1/components/speed-check/speed-check.html

```diff
@@ -1,32 +1,39 @@
-<ion-row>
+<ion-row class="speed-check-wrapper">
+  <ion-col col-32 class="competency-button-wrapper">
+    <competency-button
+      [onTap]="onTap"
+      [onPress]="onPress"
+      [ripple]="canButtonRipple()"
+      class="speed-check-competency"
+      [ngClass]="{
+        'fault': faultCount() > 0,
+        'serious-fault': hasSeriousFault(),
+        'dangerous-fault': hasDangerousFault()
+      }"
+    >
+      <driving-faults-badge class="driving-faults" [count]="faultCount()"></driving-faults-badge>
+      <span class='competency-label'>{{ getLabel() }}</span>
+      <div class="fault-wrapper">
+        <serious-fault-badge *ngIf="hasSeriousFault() || (!hasSeriousFault() && !hasDangerousFault())" [showBadge]="hasSeriousFault()"></serious-fault-badge>
```

Comment:
confused by logic on ngif.   Why is the or condition  in there?  as I guess showBadge will still hide it.

Reply:
Maybe a comment explaining the logic would be a good way to avoid confusion. Going to do that.